builtin/mainmenu/settings/components.lua

```diff
@@ -344,18 +344,16 @@ function make.flags(setting)
 			local y = 0.55
 
 			for _, possible in ipairs(setting.possible) do
-				if possible:sub(1, 2) ~= "no" then
```

Comment:
No flags should be hidden if there's a matching flag without the no

Reply:
Ideally, we'd have a function that parses flag lists and returns a table with true, false, or nil for unspecified. And a corresponding set function. To avoid "noise" being interpreted as no oise, I guess it needs to know which flags are valid.

This is "Add flag util" in https://github.com/minetest/minetest/issues/13476

Reply:
> No flags should be hidden if there's a matching flag without the no

Sure, but this is `setting.possible`, the possible flags. There should be no "no" prefixes here, this isn't the default value. This is the "type" rather than the default "value" so to speak.

From `builtin/settingtypes.txt`:

```
#   * flags:
#            Flags are always separated by comma without spaces.
#            - default possible_flags
```

It seems sometimes only a default is provided, in which case the `possible_flags` need to be inferred. *Only then* should settings with `no` prefixes be ignored (which also means that if you provide the `possible_flags` as the docs tell you to, you can have flags such as `noise`).

This is a bug in `builtin/mainmenu/settings/settingtypes.lua`, where it just defaults to `possible = default` without stripping such prefixes.